CMake/Runtime/Preprocessing.cmake

```diff
@@ -52,22 +52,34 @@ foreach (LIB ${SKETCH_LINK_LIBS})
   if (NOT TARGET "${LIB}")
     continue ()
   endif ()
-  get_target_property (LIB_INCLUDE_DIRS "${LIB}" INCLUDE_DIRECTORIES)
-  if (LIB_INCLUDE_DIRS)
-    list (APPEND INCLUDE_DIRS ${LIB_INCLUDE_DIRS})
-  endif ()
-  get_target_property (LIB_COMP_DEFS "${LIB}" COMPILE_DEFINITIONS)
-  if (LIB_COMP_DEFS)
-    list (APPEND COMP_DEFS ${LIB_COMP_DEFS})
-  endif ()
-  get_target_property (LIB_INCLUDE_DIRS "${LIB}" INTERFACE_INCLUDE_DIRECTORIES)
-  if (LIB_INCLUDE_DIRS)
-    list (APPEND INCLUDE_DIRS ${LIB_INCLUDE_DIRS})
-  endif ()
-  get_target_property (LIB_COMP_DEFS "${LIB}" INTERFACE_COMPILE_DEFINITIONS)
-  if (LIB_COMP_DEFS)
-    list (APPEND COMP_DEFS ${LIB_COMP_DEFS})
-  endif ()
+  get_target_property (target_type "${LIB}" TYPE)
+  if (target_type STREQUAL "INTERFACE_LIBRARY")
+    get_target_property (LIB_INCLUDE_DIRS "${LIB}" INTERFACE_INCLUDE_DIRECTORIES)
+    if (LIB_INCLUDE_DIRS)
+      list (APPEND INCLUDE_DIRS ${LIB_INCLUDE_DIRS})
+    endif ()
+    get_target_property (LIB_COMP_DEFS "${LIB}" INTERFACE_COMPILE_DEFINITIONS)
+    if (LIB_COMP_DEFS)
+      list (APPEND COMP_DEFS ${LIB_COMP_DEFS})
+    endif ()
```

Comment:
This is duplicated below and could be hoisted out

Reply:
i fix.